avocado/core/loader.py

```diff
@@ -321,7 +321,7 @@ def load_test(self, test_factory):
             test_path = test_parameters.pop('modulePath')
         else:
             test_path = None
-        if isinstance(test_class, str):
```

Comment:
Well it's debatable whether we have to change these as `six.string_types` is `basestring` in py2 and `str` in py3, therefor this is not switch to `string_types` but fix to us not using the right string asset. Anyway yes, the use of `string_types` is safer given we'll be doing more string encoding, which might result in new failures when not checking for `basestring`.

Reply:
Actually I just found one important difference we need to pay attention to. The encoded bytes in py2 are instance of `basestring`, but they are not instance of `str` in py3. That will be tricky...

Reply:
PS: Yes, it looks like you've tackled all `basestring/str`s